docs/user_guide.rst

```diff
@@ -48,6 +48,22 @@ Connecting with remote SSH server
    :emphasize-lines: 7-13
 
 
+Connecting over GSSAPI
+----------------------
+
+.. attention::
+
+   This requires that your libssh is compiled with GSSAPI support
+   enabled.
+
+Using GSSAPI, password or private key is not necessary, but client and
+service principals may be specified.
+
+.. literalinclude:: _samples/gssapi.py
+   :language: python
+   :lines: 11-21
```

Comment:
It may be better maintainable to use markers for such simple examples:
```suggestion
   :start-after: try:
   :end-before: except LibsshSessionException as ssh_exc:
   :emphasize-lines: 8-9
```
(I haven't tried this locally, so better test first)